src/substrate-lib/substrate-context.tsx

```diff
@@ -0,0 +1,377 @@
+import { createInterBtcApi } from '@interlay/interbtc-api';
+import { ApiPromise } from '@polkadot/api';
+import { web3Accounts, web3Enable, web3FromAddress } from '@polkadot/extension-dapp';
+import { InjectedAccountWithMeta, InjectedExtension } from '@polkadot/extension-inject/types';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { TypeRegistry } from '@polkadot/types/create';
+import jsonrpc from '@polkadot/types/interfaces/jsonrpc';
+import type { DefinitionRpcExt } from '@polkadot/types/types';
+import { keyring } from '@polkadot/ui-keyring';
+import { Keyring } from '@polkadot/ui-keyring/Keyring';
+import { isTestChain } from '@polkadot/util';
+import * as React from 'react';
+import { useLocalStorage } from 'react-use';
+
+import { APP_NAME } from '@/config/relay-chains';
+import { SELECTED_ACCOUNT_LOCAL_STORAGE_KEY } from '@/config/wallets';
+
+import * as constants from '../constants';
+
+const UNABLE_TO_INITIALISE_OPTIONS_MORE_THAN_ONCE = 'Unable to initialise options more than once';
+
+enum KeyringStatus {
+  Idle = 'IDLE',
+  Loading = 'LOADING',
+  Ready = 'READY',
```

Comment:
NIT: could be `SUCCESS`

Reply:
Just like react-query
![image](https://user-images.githubusercontent.com/43269067/191029913-2094edba-67f7-4fea-a998-d1e4ebd3fdd8.png)

Reply:
For now, I'd suggest going as it is because "READY" matches "ready" event from the polkadot-js API.